tests/client/net_handler_client.c

```diff
@@ -163,26 +163,38 @@ static int recv_tcp_data(robot_desc_field_t client, int tcp_fd) {
         return -1;
     }
 
-    // unpack the message
-    if ((msg = text__unpack(NULL, len, buf)) == NULL) {
-        fprintf(tcp_output_fp, "Error unpacking incoming message from %s\n", client_str);
-    }
-
-    // print the incoming message
-    if (msg_type == LOG_MSG) {
-        for (int i = 0; i < msg->n_payload; i++) {
-            fprintf(tcp_output_fp, "%s", msg->payload[i]);
+    if (msg_type == TIME_STAMP_MSG) {
+        TimeStamps* time_stamp_msg = time_stamps__unpack(NULL, len, buf);
+        if (time_stamp_msg == NULL) {
+            log_printf(ERROR, "recv_new_msg: Cannot unpack time_stamp msg");
+        }
+        uint64_t final_timestamp = millis();
```

Comment:
So `mills()` in runtime_util.c calls `gettimeofday`which gives us the number of seconds and microseconds since epoch. We then do some manipulation to turn that into milliseconds and send it on its way to Dawn. It seems to work from what we tested today

Reply:
Ah okay I didn't realize `millis()` was our own function, I think wrote that sometime in the summer of last year haha
But ok sounds good, glad to know it works!